### PR TITLE
Feat: added support for MongoDB as vector DB

### PR DESCRIPTION
**Pull request by marcusschiesser**
- compatible with Python 0.9
- added ingestion and query from https://github.com/run-llama/mongodb-demo as example
- updated SimpleMongoReader to have the same features as Python 0.9 version

### CODE REVIEW

File: apps/mongodb/README.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+Now you're creating a vector search client for Mongo. In addition to a MongoDB client object, you again tell it what database everything is in. This time you give it the name of the collection where you'll store the vector embeddings, and the name of the vector search index you'll create in the next step.
+
+### Create a vector search index
+
+Now if all has gone well you should be able to log in to the Mongo Atlas UI and see two collections in your database: the original data in `tiny_tweets_collection`, and the vector embeddings in `tiny_tweets_vectors`.
+
+![MongoDB Atlas collections](./docs/3_vectors_in_db.png)
+
+Now it's time to create the vector search index so that you can query the data.
+It's not yet possible to programmatically create a vector search index using the [`createIndex`](https://www.mongodb.com/docs/manual/reference/method/db.collection.createIndex/) function, therefore we have to create one manually in the UI.
```
Comment (marcusschiesser) [inline apps/mongodb/README.md:83]: @yisding creating the index programmatically is not yet possible

File: apps/mongodb/package.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (marcusschiesser) [inline apps/mongodb/package.json:1]: @yisding i saw that there are examples in `apps` and `examples` - where is the better place for this example?

File: packages/core/src/storage/vectorStore/utils.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+  return metadata;
+}
+
+export function metadataDictToNode(metadata: Metadata): BaseNode {
+  const nodeContent = metadata["_node_content"];
+  if (!nodeContent) {
+    throw new Error("Node content not found in metadata.");
+  }
+  const nodeObj = JSON.parse(nodeContent);
```
Comment (marcusschiesser) [inline packages/core/src/storage/vectorStore/utils.ts:47]: @yisding it works to reconstruct objects from Python 0.9, but there is an issue that the node attributes in Python are snake_case and in TS they are camelCase, so a reconstructed `TextNode` in TS (coming from Python) looks like this:

```
_TextNode {
      id_: '95d3f69c-4197-41ca-9f87-b78d9892867f',
      metadata: {},
      excludedEmbedMetadataKeys: [],
      excludedLlmMetadataKeys: [],
      relationships: [Object],
      hash: '4JN1LIB89VS6Hq1tZbtoYrJDCN0uLyHs68Et8K4B/ZA=',
      embedding: null,
      excluded_embed_metadata_keys: [],
      excluded_llm_metadata_keys: [],
      text: "@bradleymeck @aredridel @nebrius @addaleax All fair points. We couldn't figure out a way to get around HTTP so we think bundles will still be used as a perf optimization, but getting away without a build step at development time is a worthwhile win, I think.",
      start_char_idx: null,
      end_char_idx: null,
      text_template: '{metadata_str}\n\n{content}',
      metadata_template: '{key}: {value}',
      metadata_seperator: '\n',
      class_name: 'TextNode',
      metadataSeparator: '\n'
    }
    ```
    I think we should either use snake_case in TS too or add a mapping

**Review by yisding — APPROVED** [review]
Thanks!

File: packages/core/src/storage/vectorStore/utils.ts (hunk 3 above)
Comment (yisding) [reply packages/core/src/storage/vectorStore/utils.ts:47]: Let's create a TS class to track and convert for now. The interface on the python side may also not be entirely stable yet, so let's create one that says TextNodev09 etc.

**Pull request merged**